reya_trading/auth/signatures.py

```diff
@@ -0,0 +1,332 @@
+        
+        # ABI encode the inputs: ['int256', 'uint256'] → [size, price]
+        # Size needs to be scaled to E18 and signed
+        # Price needs to be scaled to E18
+        scaler = self.scale(18)
+        size_e18 = scaler(size)
+        price_e18 = scaler(price)
+        
+        inputs_encoded = encode(
+            ['int256', 'uint256'],
+            [size_e18, price_e18]
+        )
+        
+        # Determine order type
+        order_type = 4 if reduce_only else 3  # REDUCE_ONLY_MARKET_ORDER : MARKET_ORDER
+        
+        # Get counterparty account IDs (pool account ID)
+        pool_account_id = 2 if self._chain_id == 1729 else 2  # TODO: verify testnet value
+        
+        return self.sign_orders_gateway_order(
+            account_id=account_id,
+            market_id=market_id,
+            exchange_id=2,  # REYA_DEX_ID
+            counterparty_account_ids=[pool_account_id],
+            order_type=order_type,
+            inputs=inputs_encoded.hex(),
+            deadline=deadline,
+            nonce=nonce
+        )
+    
+    def sign_conditional_order(
+        self,
+        market_id: int,
+        order_type: ConditionalOrderType,
+        is_buy: bool,
+        trigger_price: Union[str, float],
+        nonce: int,
+        order_base: Union[str, float] = 0,
+        order_price_limit: Optional[Union[str, float]] = None,
+    ) -> str:
+        """
+        Sign a conditional order (limit, take profit, stop loss) using EIP-712.
+        
+        Args:
+            market_id: The market ID for this order
+            order_type: The type of conditional order
+            is_buy: Whether this is a buy order
+            trigger_price: Price at which the order triggers
+            order_base: Base amount of the order
+            order_price_limit: Limit price for the order
+            nonce: Random nonce
+            deadline: Signature expiration timestamp
+            
+        Returns:
+            Hex-encoded signature
+        """
+
+        deadline = 10 ** 18  # CONDITIONAL_ORDER_SIG_DEADLINE
+        
+        # Encode inputs based on order type
+        inputs = self.encode_inputs(
+            order_type, is_buy, trigger_price, order_base, order_price_limit
+        )
+        
+        # Get counterparty account IDs (pool account ID)
+        # For mainnet: 2, for testnet: might be different
+        pool_account_id = 2 if self._chain_id == 1729 else 2  # TODO: verify testnet value
+        
+        return self.sign_orders_gateway_order(
+            account_id=self.config.account_id,
+            market_id=market_id,
+            exchange_id=2,  # REYA_DEX_ID
+            counterparty_account_ids=[pool_account_id],
+            order_type=order_type.value,
+            inputs=inputs.hex(),
+            deadline=deadline,
+            nonce=nonce
+        )
+    
+    def sign_cancel_order(self, order_id: str) -> str:
```

Comment:
once the leverage endpoint is made private, we'll need to also add signing support for that here